does your code works on windows?

iam quite a good developer on nodejs
but the libraries r not compiling, some have missing header files and some misplaced files

in the src/native

could not compile, does it work on windows 10?
or linux only

errors iam getting are

..\src\native\sha256.c(136): warning C4244: '=': conversion from 'unsigned __int64' to 'BYTE', possible loss of data [E:\angularpractise\core\build\nimiq_node.vcxproj]
..\src\native\sha256.c(137): warning C4244: '=': conversion from 'unsigned __int64' to 'BYTE', possible loss of data [E:\angularpractise\core\build\nimiq_node.vcxproj]
..\src\native\sha256.c(138): warning C4244: '=': conversion from 'unsigned __int64' to 'BYTE', possible loss of data [E:\angularpractise\core\build\nimiq_node.vcxproj]
..\src\native\sha256.c(139): warning C4244: '=': conversion from 'unsigned __int64' to 'BYTE', possible loss of data [E:\angularpractise\core\build\nimiq_node.vcxproj]
..\src\native\sha256.c(140): warning C4244: '=': conversion from 'unsigned __int64' to 'BYTE', possible loss of data [E:\angularpractise\core\build\nimiq_node.vcxproj]
..\src\native\sha256.c(141): warning C4244: '=': conversion from 'unsigned __int64' to 'BYTE', possible loss of data [E:\angularpractise\core\build\nimiq_node.vcxproj]
..\src\native\sha256.c(142): warning C4244: '=': conversion from 'unsigned __int64' to 'BYTE', possible loss of data [E:\angularpractise\core\build\nimiq_node.vcxproj]
  fe.c
  ge.c
  keypair.c
  memory.c
  sc.c
  sha512.c
  sign.c
  verify.c
e:\angularpractise\core\src\native\sha256.h(20): error C2371: 'WORD': redefinition; different basic types (compiling source file ..\src\native\nimiq_native.c) [E:\angularpractise\core\build\nimiq_node.vcxproj]
  C:\Program Files (x86)\Windows Kits\8.1\Include\shared\minwindef.h(159): note: see declaration of 'WORD' (compiling source file ..\src\native\nimiq_native.c)

--- a/binding.gyp
+++ b/binding.gyp
@@ -21,6 +21,9 @@
                 "src/native/ed25519/verify.c",
                 "src/native/nimiq_node.cc"
             ],
+            "defines": [
+                "ARGON2_NO_THREADS"
+            ],
             "include_dirs": [
                 "<!(node -e \"require('nan')\")",
                 "src/native"
@@ -28,14 +31,12 @@
             "cflags_c": [
                 "-std=c99",
                 "-march=native",
-                "-mtune=native",
-                "-DARGON2_NO_THREADS"
+                "-mtune=native"
             ],
             "xcode_settings": {
                 "OTHER_CFLAGS": [
                     "-march=native",
-                    "-mtune=native",
-                    "-DARGON2_NO_THREADS"
+                    "-mtune=native"
                 ]
             }
         }

--- a/src/native/sha256.c
+++ b/src/native/sha256.c
@@ -29,7 +29,7 @@
 #define SIG1(x) (ROTRIGHT(x,17) ^ ROTRIGHT(x,19) ^ ((x) >> 10))
 
 /**************************** VARIABLES *****************************/
-static const WORD k[64] = {
+static const WORD32 k[64] = {
 	0x428a2f98,0x71374491,0xb5c0fbcf,0xe9b5dba5,0x3956c25b,0x59f111f1,0x923f82a4,0xab1c5ed5,
 	0xd807aa98,0x12835b01,0x243185be,0x550c7dc3,0x72be5d74,0x80deb1fe,0x9bdc06a7,0xc19bf174,
 	0xe49b69c1,0xefbe4786,0x0fc19dc6,0x240ca1cc,0x2de92c6f,0x4a7484aa,0x5cb0a9dc,0x76f988da,
@@ -43,7 +43,7 @@ static const WORD k[64] = {
 /*********************** FUNCTION DEFINITIONS ***********************/
 void sha256_transform(SHA256_CTX *ctx, const BYTE data[])
 {
-	WORD a, b, c, d, e, f, g, h, i, j, t1, t2, m[64];
+	WORD32 a, b, c, d, e, f, g, h, i, j, t1, t2, m[64];
 
 	for (i = 0, j = 0; i < 16; ++i, j += 4)
 		m[i] = (data[j] << 24) | (data[j + 1] << 16) | (data[j + 2] << 8) | (data[j + 3]);
@@ -98,7 +98,7 @@ void sha256_init(SHA256_CTX *ctx)
 
 void sha256_update(SHA256_CTX *ctx, const BYTE data[], size_t len)
 {
-	WORD i;
+	WORD32 i;
 
 	for (i = 0; i < len; ++i) {
 		ctx->data[ctx->datalen] = data[i];
@@ -113,7 +113,7 @@ void sha256_update(SHA256_CTX *ctx, const BYTE data[], size_t len)
 
 void sha256_final(SHA256_CTX *ctx, BYTE hash[])
 {
-	WORD i;
+	WORD32 i;
 
 	i = ctx->datalen;
 

--- a/src/native/sha256.h
+++ b/src/native/sha256.h
@@ -17,13 +17,13 @@
 
 /**************************** DATA TYPES ****************************/
 typedef unsigned char BYTE;             // 8-bit byte
-typedef unsigned int  WORD;             // 32-bit word, change to "long" for 16-bit machines
+typedef unsigned int  WORD32;           // 32-bit word, change to "long" for 16-bit machines
 
 typedef struct {
 	BYTE data[64];
-	WORD datalen;
+	WORD32 datalen;
 	unsigned long long bitlen;
-	WORD state[8];
+	WORD32 state[8];
 } SHA256_CTX;
 
 /*********************** FUNCTION DECLARATIONS **********************/

